server/test/bookingTest.js

```diff
@@ -1,89 +1,192 @@
-    let res = await chai.request(server).post(tmp).send({lecture_id: newBooking.lecture_id});
+    let res = await agent.post(tmp).send({ lecture_id: newBooking.lecture_id });
     should.exist(res);
     res.body.should.be.an('object');
-    return res.status;
-        
+    return res;
+
 }
 
-const deletion =async function(){
-    const newBooking = { lecture_id: 1, student_id: 2};
-        const tmp = `/api/students/${newBooking.student_id}/lectures/${newBooking.lecture_id}`;
-        let res = await chai.request(server).delete(tmp).send();
-        should.exist(res);
-        res.body.should.be.an('object');
-        return res.status;
+const deletion = async function () {
+    let credentials = { email: data.students[0].email, password: data.students[0].password };
+    const agent = chai.request.agent(server);
+    let resLogin = await agent.post(`/api/login`).send(credentials);
+
+    let student_id = resLogin.body.id;
+    // console.log(student_id);
+
+    const newBooking = { lecture_id: 1, student_id: student_id };
+    const tmp = `/api/students/${newBooking.student_id}/lectures/${newBooking.lecture_id}`;
+    let res = await agent.delete(tmp).send();
+    should.exist(res);
+    res.body.should.be.an('object');
+    return res.status;
 
 }
 
 describe('Booking routes', function () {
-    before('create tables and clear db', async function() {
+    before('create tables and clear db', async function () {
         await dbUtils.reset();
     });
-
-    after('clear db', async function() {
-        await dbUtils.reset({ create: false });
+    before('populate db', async () => {
+        data = await dbUtils.populate();
     });
 
-    it('should allow student to book a lecture', async function() {
-        let res=await book();
-        res.should.be.equal(201);
+    it('should allow student to book a lecture', async function () {
+        let credentials = { email: data.students[0].email, password: data.students[0].password };
+        const agent = chai.request.agent(server);
+        let resLogin = await agent.post(`/api/login`).send(credentials);
+        let start_date = moment(new Date(2020, 10, 28, 16, 30)).format('YYYY-MM-DD');
+        let end_date = moment(new Date(2020, 10, 31, 16, 30)).format('YYYY-MM-DD');
+
+        
+        const tmp = `/api/students/${resLogin.body.id}/lectures`;
+        
+        let res = await agent.get(tmp).query({ from: start_date, to: end_date });
+        // console.log('BEFORE BOOKING:', res.body)
+        
+        res = await book();
+        res.should.have.property('status', 201);
+        
+        res = await agent.get(tmp).query({ from: start_date, to: end_date });
+
+        // console.log('AFTER BOOKING:', res.body)
+
     });
 
-    it('should not allow student to book a lecture already booked', async function() {
-        let res=await book();
-        res.should.not.be.equal(201);
+    it('should not allow student to book a lecture already booked', async function () {
```

Comment:
We should take a second look to this test:

-the login is duplicated (since it is already done in the book function)
-startdate and enddate are not used (the book function has its own parameters)